Bugfix/7873 time conductor input validation

davetsay opened the pull request:
<!--- Note: Please open the PR in draft form until you are ready for active review. -->
Closes #7873 

### Describe your changes:
* Checks validation of inputs on `input` event
* Only reports validation of inputs on `change` event, so user is not punished early
* Only reports logical validation errors on submit (limit out of bounds, start bounds greater than end bounds, etc.)

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [ ] Is this a [notable change](../docs/src/process/release.md) that will require a special callout in the release notes? For example, will this break compatibility with existing APIs or projects that consume these plugins?

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?

shefalijoshi commented on src/plugins/timeConductor/TimePopupFixed.vue:
```diff
@@ -11,18 +11,19 @@
           aria-label="Start date"
-          @input="validateAllBounds('startDate')"
+          @input="onInput('startDate')"
+          @change="onChange('startDate')"
```
We should use something more descriptive than `onInput` and `onChange`.

shefalijoshi reviewed (approved):
Small change but looks solid

akhenry commented on src/plugins/timeConductor/TimePopupFixed.vue:
```diff
@@ -215,96 +232,98 @@ export default {
+      input.setCustomValidity('');
+      input.title = '';
+    },
+    onInput(refName) {
```
Please give this a name that describes what it does, not the event that it handles. eg. `clearValidationAndRevalidate`

akhenry commented on src/plugins/timeConductor/TimePopupFixed.vue:
```diff
@@ -215,96 +232,98 @@ export default {
+      this.clearAllValidation();
+      this.validateInput(refName);
+    },
+    onChange(refName) {
```
Is this needed? It's just calling `handleValidationResults`

akhenry commented:
Tested this locally and it looks really good, very nice work. I noticed a _very_ minor niggle that could be fixed in a followup -


https://github.com/user-attachments/assets/c6c713e1-a76e-47e2-82a8-35a724fd8f88

davetsay commented:
> Tested this locally and it looks really good, very nice work. I noticed a _very_ minor niggle that could be fixed in a followup. Invalid time bounds result in the submit button being disabled which is great. However an invalid combination of dates leaves the submit button _enabled_. If you try and click the enabled submit button it shows a validation error but for consistency it would be great if invalid date bounds also disabled the submit button, like invalid times do
> 
> eg.
> 
>  minor.TC.bug.mov

This is by design. The point is to not punish the user early. Give them a chance to input valid bounds, and if they don't, prevent them from submitting invalid bounds. This comes into play when changing both start and end bounds. The user almost always edits the start bounds first, and then is punished too early if say they are moving the start bounds and end bounds forward in time to a time window that is after the end bounds.

I think this is what the video is showing. If not, we can re-examine.

davetsay commented:
@akhenry , @shefalijoshi , addressed your comments in this commit, https://github.com/nasa/openmct/pull/7886/commits/8d4d57d0ceb1edfce2fe2ac8835dd305135fb8b5. ready for re-review

akhenry commented:
@davetsay  Love it, the validation is subtle and brilliant. For all intents and purposes it's art.
